### PR TITLE
Add field filter support.

### PR DESCRIPTION
**Pull request by jpbetz**
This splits out the data access layer of auger into it's own package as part of this to keep the code organized. The data access layer now includes 'filter' support that uses the same field naming as already introduced --template flag.  E.g. to filter out objects in a specific namespace, one could now do:

```
auger extract -d <data-file> --filter=".Value.metadata.namespace=kube-system"
```

### CODE REVIEW

**Comment by jpbetz** [comment]
@wenjiaswe Would you give this a quick review?

File: pkg/data/data_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		{
+			name:         "prefixfilter",
+			filters:      []Filter{NewPrefixFilter("/registry/jobs")},
+			expectedKeys: []string{"/registry/jobs/default/pi"},
+		},
+		{
+			name:         "namespacefilter",
+			filters:      []Filter{mustBuildFilter(&FieldConstraint{lhs: ".Value.metadata.namespace", op: Equals, rhs: "default"})},
+			expectedKeys: []string{"/registry/jobs/default/pi", "/registry/pods/default/pi-dqtsw"},
+		},
```
Comment (wenjiaswe) [inline pkg/data/data_test.go]: From the logic of the code, there could be both prefix filter and some other filter coexisting, would it worth to add a test with both?

Comment (jpbetz) [reply pkg/data/data_test.go]: Yes, this is needed. Added.

File: pkg/data/data.go [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+		return err
+	}
+	return nil
+}
+
+// ParseFilters parses a comma separated list of '<field>=<value>' filters where each field is
+// specified in golang template format, e.g. '.Value.metadata.namespace'.
+func ParseFilters(filters string) ([]Filter, error) {
+	// TODO: This is a stopgap implementation. We need a real grammar (text/scanner + hand written
+	// parser combinator?) to generalize.
```
[the remark this answers is not in the document]
Comment (jpbetz) [reply pkg/data/data.go]: Yeah, this is sufficient for now. I'll drop the TODO. We can track any future improvements we want to make separately.

**Comment by jpbetz** [comment]
> Looks great overall, just few minor things. It would also be good to add the filter usage to the https://github.com/jpbetz/auger/blob/master/README.md, this is very useful. It can be in different PRs, it's your call:) Thanks!

Thanks for the review! I've added a section to the README as well.

**Pull request merged**